### DISCUSSION TITLE
Possibly incorrect use of a constant?

### DISCUSSION THREAD

**Discussion by homotechsual** [facebook/docusaurus#10058] [General] [answered]
In the `docusaurus clear` code we reference `GENERATED_FILES_DIR_NAME` which could be a relative or absolute directory path according to the [code comments](https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-utils/src/constants.ts#L43).

The reference in `clear.ts` assumes it's always a relative path.

```js
  const generatedFolder = {
    path: path.join(siteDir, GENERATED_FILES_DIR_NAME),
    description: 'generated folder',
  };
  ```

**Comment by slorber** [comment, marked as answer]
Hmmm yes I can confirm this command does not work:

```bash
DOCUSAURUS_GENERATED_FILES_DIR_NAME=/Users/sebastienlorber/Desktop/projects/docusaurus/website/.docusaurus yarn clear
```

We should probably use `path.resolve` instead of `path.join` in those cleanup paths (same for build folder)

Do you want to send a PR?

**Reply by homotechsual** [reply]
Will do - thanks for the confirmation.
